app/controllers/AdditionalMetadataClosureStatusController.scala

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import auth.TokenSecurity
+import configuration.KeycloakConfiguration
+import controllers.util.InputNameAndValue
+import controllers.util.MetadataProperty.{clientSideOriginalFilepath, closureType}
+import graphql.codegen.types.{FileFilters, UpdateFileMetadataInput}
+import org.pac4j.play.scala.SecurityComponents
+import play.api.cache.AsyncCacheApi
+import play.api.data.Form
+import play.api.data.Forms.{boolean, mapping}
+import play.api.mvc.{Action, AnyContent, Request, Result}
+import services.{ConsignmentService, CustomMetadataService}
+
+import java.util.UUID
+import javax.inject.Inject
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+class AdditionalMetadataClosureStatusController @Inject()(val consignmentService: ConsignmentService,
+                                                          val customMetadataService: CustomMetadataService,
+                                                          val keycloakConfiguration: KeycloakConfiguration,
+                                                          val controllerComponents: SecurityComponents,
+                                                          val cache: AsyncCacheApi
+                                                         ) extends TokenSecurity {
+
+
+  val closureStatusForm: Form[ClosureStatusFormData] = Form(
+    mapping(
+      "closureStatus" -> boolean
+        .verifying("You must confirm this closure has been approved before continuing.", b => b)
+    )(ClosureStatusFormData.apply)(ClosureStatusFormData.unapply)
+  )
+
+  val closureStatusField: InputNameAndValue = InputNameAndValue("closureStatus", "Yes, I confirm")
+
+
+  def getClosureStatusPage(consignmentId: UUID, fileIds: List[UUID]): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+
+    val filters = Option(FileFilters(None, Option(fileIds), None))
+
+    for {
+      details <- consignmentService.getConsignmentDetails(consignmentId, request.token.bearerAccessToken)
+      consignment <- consignmentService.getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, filters)
+      response <- if (consignment.files.nonEmpty) {
+        val filePaths = consignment.files.flatMap(_.fileMetadata).filter(_.name == clientSideOriginalFilepath).map(_.value)
+        val closureStatus = consignment.files.flatMap(_.fileMetadata).find(_.name == closureType.name).exists(_.value == closureType.value)
+        cache.set(s"$consignmentId-data", (consignment.consignmentReference, filePaths, details.parentFolderId.get), 1.hour)
```

Comment:
Should the cache be checked first to see if it exists? If so use the data from the cache, if not, only then make the calls to the API and set it?

Reply:
Yeah, I thought the same but then I realized if we cache it then it won't show the updated page if the user comes back. It should show the checkbox as checked as the user is already confirmed.
If we really want to go with cache then we have to update the cache when the user confirms it but I don't think it's a good idea. What do you think?

Reply:
Yeah, good point. Lets go without the cache. We can always revisit if needed.